backend/Dockerfile

```diff
@@ -15,7 +15,7 @@ COPY backend/requirements.txt .
 RUN python3 -m pip install -r requirements.txt
 
 # Downloading Argo CLI so that the samples are validated
-#ADD  https://github.com/argoproj/argo/releases/download/v2.7.5/argo-linux-amd64 /usr/local/bin/argo
+#ADD  https://github.com/argoproj/argo/releases/download/v2.11.6/argo-linux-amd64 /usr/local/bin/argo
```

Comment:
Not sure if I need to update this one

Reply:
I guess it is good to update the comment but not sure how this will be used.

Reply:
I think we should leave this one as well since it is the Kubeflow backend which will be running 2.7.7. Argo is in this case used to validate the workflows to check if they are valid according to argo.

Reply:
It seems that we have several places that use Argo but they can be somehow independent of each other.  Can you tell me how each of them is used so that I can get a clearer picture of how they work together?